clove/network/ethereum/transaction.py

```diff
@@ -155,23 +156,22 @@ def __init__(
         self.contract = None
         self.locktime = None
         self.locktime_unix = None
+        self.token_value_base_units = 0
+        self.value_base_units = 0
 
         self.set_locktime()
         self.set_secrets()
 
         if self.token:
             self.token_address = self.token.token_address
             self.symbol = self.token.symbol
-            self.contract_address = self.token.contract_address
-            self.abi = self.token.abi
-            self.value_base_units = self.token.value_to_base_units(self.value)
-            self.set_token_contract()
+            self.token_value_base_units = self.token.value_to_base_units(self.value)
         else:
+            self.token_address = '0x0000000000000000000000000000000000000000'
```

Comment:
This empty address is really required for non-tokens?

Reply:
We need to pass something that will match the type of token address argument from ABI.